### ISSUE TITLE
Server Crash Spell ID 120360 label:Crash 

### ISSUE THREAD

**Issue by bommel63** [AshamaneProject/AshamaneCore#374]
Server Crash Bug Spell ID 120360

http://www.wowhead.com/spell=120360/barrage

Hunter Talent Spell ID 120360  crashes the server.

**Comment by zgn1988** [comment]
```
      void HandleOnHit()
        {
            Player* player = GetCaster()->ToPlayer();
            Unit* target = GetHitUnit();

            if (!player || !target)
                return;
```

bad code

**Comment by zgn1988** [comment]
```
  void HandleOnHit()
      {
          Player* player = GetCaster()->ToPlayer();
         if  (Unit* target = GetHitUnit())
         {
///////////////
///////////
/////////
         };


```

**Comment by Traesh** [comment]
Duplicate of #370, already fixed by e6ce426c9a1cb4c8653c1a2c7d202bc4db705f7a
